### PR TITLE
Nimble numeric matchers support some c-types

### LINKED ISSUE #113: Comparing booleans in obj-c requires NSNumber boxing
`expect(@YES).to(beTrue());` requires `@YES` instead of just `YES`, otherwise you'll get "Implicit conversion of 'BOOL' (aka 'bool') to 'id' is disallowed with ARC".

### LINKED ISSUE #254: Is there a decent way to compare NSRanges in Objective-C
Currently you have to write something super clumsy like:

`expect([NSValue valueWithRange:feature.range]).to(equal([NSValue valueWithRange:NSMakeRange(3, 5)]));`

Is there a better way?

### PR DESCRIPTION
**Pull request by jeffh**
- Add (non-)nullability for Nimble's Objective-C Headers
- Convert assertion to optional

Example:

```objc
// objc
expect(2).to(equal(2));
expect(98.7).to(beCloseTo(98.7));
expect(2).to(beLessThan(100));
expect(@[@1, @2]).to(haveCount(2));
```

# TODO

- [x] Needs Documentation
- [x] Fix #113 
- [x] Fix #254

### CODE REVIEW

**Comment by jeffh** [comment]
@Quick/core, this is ready for review.

File: README.md [hunk 1]
```diff
@@ -414,6 +415,17 @@ to keep in mind when using Nimble in Objective-C:
 
    expect(@(1 + 1)).to(equal(@2));
    expect(@"Hello world").to(contain(@"world"));
+
+   // Boxed as NSNumber*
+   expect(2).to(equal(2));
+   expect(1.2).to(beLessThan(2.0));
+   expect(true).to(beTruthy());
+
+   // Boxed as NSString *
```
Comment (modocache) [inline README.md]: Tiny nit-pick: Above you use `NSNumber*` (no space), here you use `NSString *` (space between type and pointer).

File: README.md [hunk 2]
```diff
@@ -426,6 +438,28 @@ to keep in mind when using Nimble in Objective-C:
...
+- `beLessThan`
+- `beLessThanOrEqual`
+- `beCloseTo`
+- `beTrue`
+- `beFalse`
+- `beTruthy`
+- `beFalsy`
+- `haveCount`
+
+If you would like to see more, [file an issue](https://github.com/Quick/Nimble/issues).
```
Comment (modocache) [inline README.md]: Awesome docs! 😍

**Comment by mkauppila** [comment]
Seriously cool stuff. Good job @jeffh IMHO it's ready to be merged :)

**Pull request merged**